core/api/src/app/payments/update-pending-payments.ts

```diff
@@ -129,26 +129,32 @@ const updatePendingPayment = wrapAsyncToRunInSpan({
     pendingPayment: LedgerTransaction<WalletCurrency>
     logger: Logger
   }): Promise<true | ApplicationError> => {
-    const { paymentHash, pubkey, type: txType, journalId } = pendingPayment
-    addAttributesToCurrentSpan({ walletId, paymentHash, txType })
-
-    const paymentLogger = logger.child({
-      topic: "payment",
-      protocol: "lightning",
-      transactionType: "payment",
-      onUs: false,
-      payment: pendingPayment,
-    })
-
-    const lndService = LndService()
-    if (lndService instanceof Error) return lndService
-
+    const { paymentHash, pubkey, type: txType } = pendingPayment
     // If we had PaymentLedgerType => no need for checking the fields
-    if (!paymentHash)
+    if (!paymentHash) {
       return new InconsistentDataError("paymentHash missing from payment transaction")
-    if (!pubkey)
+    }
+    if (!pubkey) {
       return new InconsistentDataError("pubkey missing from payment transaction")
+    }
 
+    addAttributesToCurrentSpan({
+      "payment.walletId": walletId,
+      "payment.request.paymentHash": paymentHash,
+      "payment.txType": txType,
+    })
+
+    const persistedLookup = await LnPaymentsRepository().findByPaymentHash(paymentHash)
+    if (persistedLookup instanceof Error) return persistedLookup
```

Comment:
I am not sure about this, LnPaymentsRepository is not the source of truth and previously we had sync issues with that collection.  In previous code is used only after a timeout

Reply:
cc @nicolasburtey @bodymindarts

Reply:
So far it's purely used for tracing attributes purposes, not critical to the use-case. Used to fetch `isExpired` for the htlc bug edge case and `destination` for context on successful and failed payments. Both can be undefined so the the only issue would be if we think the lookup could somehow wrongly return a `CouldNotFind` error which would early-return the function

Reply:
Added handling for case of `CouldNotFind` issue so that it does not stop the use-case https://github.com/GaloyMoney/galoy/pull/4156/commits/7655eead436bc2107c5ef71cec779fad3fc2aaa5